### ISSUE TITLE
Ep 05

### ISSUE THREAD

**Issue by josephmje** [jerdra/scwg2018_python_neuroimaging#22]
#### Pandas concepts to move to intro dataset notebook
- [x] read_csv
- [x] head
- [x] selecting columns
- [ ] creating a new column
- [x] row indexing
***
- [ ] Include information about what exactly is a motion confounder and the types of motion confounders.
- [x] Visualize how time series changes depending on whether you apply a low pass filter, high pass filter, detrending, standardize.
- [x] Is TR ever explained anywhere?
#### Setting up Motion Estimates
- [x] Fix typo "language-python"
- [x] More explanation about each column being a confounder, and the fact that you have to select which ones to use rather than using all of them.
#### The Yeo 2011 Schema
- [ ] Dummy TR -- should visualization of selecting a voxel then looking at time series showing intensity drop off.
#### Setting up Confound Variables for Regression
- [ ] Then make an exercise for selecting the columns that we want to use for confound regression.
#### Calculating Temporal Derivatives
- [x] Find a better way to do this. Get a better explanation, instead of **temporal derivatives**, talk about change in confound value over time.
#### Dummy TR Drop
- [ ] Make this an exercise instead, how do we pick volumes after the 5th frame?
#### Applying Confound Regression
- [x] Skip the second part about using nilearn.signal.clean this is no longer relevant as relevant with newer versions of nibabel.
#### Exercise Idea?
- [ ] Applying confound regression to a new data image, fill in the blanks style?

**Comment by josephmje** [comment]
Move teaching pandas to "intro to dataset" notebook.
I'm working on an ipywidget for visualizing time series changes. Will submit a WIP PR for this soon.

**Comment by jerdra** [comment]
For explaining what a TR is, this is something that should be explained in the Nibabel section or the intro to nilearn section. I'll update accordingly to deal with 4D images and how we can think about them.

**Comment by josephmje** [comment]
maybe nilearn since i've only got T1s in nibabel

**Comment by jerdra** [comment]
sure, it does feel like it fits more naturally in the nibabel section (after talking about 3D arrays before visualization), but I think reducing the density of intro content would be good.

**Comment by josephmje** [comment]
ahhh yea i see what you mean. i do mention the 4th dimension as time when describing the shape of the array. maybe we can do a call back to that part

**Comment by jerdra** [comment]
done implementing in nilearn notebook, i just used the same fig and extended it to the 4th dimension, think it should be okay!

**Comment by jerdra** [comment]
checking off python-language because that's data carpentry specific..

**Comment by jerdra** [comment]
#27 contains updates to better explain confound_df columns

**Comment by jerdra** [comment]
#27 has small explanation of temporal derivatives using X position confound and "X velocity" as example

**Comment by jerdra** [comment]
Re: "Applying confound regression", I think we can leave it up there and just skip it in the notebook

**Comment by jerdra** [comment]
Re: "Exercises", let's do the exercise at the very beginning of the functional connectivity section? Thoughts?

**Comment by josephmje** [comment]
Don't think I have row indexing in pandas yet. Should that also be added to the dataset notebook?

**Comment by josephmje** [comment]
Yup, I like the idea of applying it to the FC notebook. Would be a nice way to recap everything.

**Comment by jerdra** [comment]
if there's a way to make it part of working with the data that would be great, if not i don't mind explaining it in the signal cleaning section. Just let me know so I can work on it asap :+1:

Yeah ideally! But there's also the question of whether we just want to jump to NiftisLabelsMasker instead of using image.clean again? We might have to remove first bit for timing

**Comment by josephmje** [comment]
I think I can make row indexing work!

True, should we keep image.clean in there but then skip to NiftiLabelsMasker if we notice we're running low on time?

**Comment by jerdra** [comment]
Sure I think that's reasonable!

**Comment by jerdra** [comment]
Need to use pybids here, will make a PR in a bit

**Comment by jerdra** [comment]
done
